lib/transaction.py

```diff
@@ -349,10 +349,8 @@ def deserialize(raw):
     start = vds.read_cursor
     d['version'] = vds.read_int32()
     n_vin = vds.read_compact_size()
-    assert n_vin != 0
```

Comment:
Why remove this assertion? I don't get it.. Can you elaborate?

Reply:
In theory we may want to deserialize off-chain transactions with 0 inputs or 0 outputs. For example, this was happening in the SLP unit tests.

In particular I was inspired by looking at the bitcoind RPC recently, it has an RPC called `fundrawtransaction` that lets you pass in 0-input raw transactions and it appends inputs to them.

Reply:
I can't think of a use for a 0-output transaction but we can allow it 'for free'.

Reply:
Aha I see. Yeah ok that makes sense.